### ISSUE TITLE
Empty bibliography in HTML output (Origin: bugzilla #682602)

### ISSUE THREAD

**Issue by doxygen** [doxygen/doxygen#4860]
status RESOLVED severity *major* in component *general* for *---*
Reported in version *unspecified* on platform *Other*
Assigned to: Dimitri van Heesch

On 2012-08-24 12:35:53 +0000, Philipp Moeller wrote:

> Given the file foobar.h
> 
>     /*!
>     \cite cgalafh-pdecm-02
>     */
>     class foobar {};
> 
> and the file biblio.bib
> 
>     @article{bibentry,
>         author =  "F. O. Obar",
>         title =   "Test",
>         journal = "Test"
>     }
> 
> with a default generated doxygen configuration where
> 
>     CITE_BIB_FILES         = biblio.bib
> 
> is set.
> 
> The warning
> 
>     Generating docs for page citelist...
>     citelist:3: warning: expected <dt> tag but found <dl> instead!
>     citelist:4: warning: unexpected end of comment while looking for a html description title
> 
> 
> is generated and the resulting "Bibliographic References" page is empty.
> 
> Versions:
> 
>     $ bibtex --version
>     BibTeX 0.99d (TeX Live 2012/Arch Linux)
>     kpathsea version 6.1.0
> 
>     $ doxygen --version
>     1.8.2

On 2012-08-27 13:25:29 +0000, Philipp Moeller wrote:

> After some more investigation: My first analysis was wrong. The error occurs when an OUTPUT_DIRECTORY is specified.
> 
> The problem seems to be causee when the perl script bib2xhtml has to deal with a citelist.doc file in another directory.

On 2012-09-22 19:30:56 +0000, Dimitri van Heesch wrote:

> Confirmed. Should be fixed in the next subversion update.

On 2012-10-01 11:49:59 +0000, Philipp Moeller wrote:

> I cannot confirm that this is fixed with the latest update. The problem is still there when I specify an output directory. Also, please note the mistake in my original bug report where the documentation for class foobar should be \cite bibentry

On 2012-10-07 18:28:50 +0000, Dimitri van Heesch wrote:

> For me it works with the above example (and the \cite fix). Can you please attach a self-contained example (source+config file in a tar or zip) for which you tested it contains the problem so that I can reproduce the problem?

On 2012-10-18 15:00:39 +0000, Philipp Moeller wrote:

> I just confirmed that this works. Something on my side went wrong.
> 
> A little quirk is that the temporary files don't get removed anymore due to commented lines starting at 296 cite.cpp. I guess that is an oversight and not intended.

**Comment by hermannkleier** [comment]
Alas, the problem is **not** completed: Arch Linux as of 2024-10-28, doxygen 1.12.0, BibTeX 0.99d, Linux 6.11.5-arch1-1, …
I tripped over the same symptoms when I tried to start up a tiny “project”. After copying _all_ files to another computer, the problem vanished. Checking the environment was to no avail. In the end, the solution was to install all packages from texlive. Then the error was gone.
Bottom line: Document the missing dependencies.

**Comment by albert-github** [comment]
@hermannkleier 
The packages used probably also depend on the texts you are using and the settings there within and thus the packages that have to be installed.
- which packages are you missing?

Will be hard to reproduce, but maybe you have a small example that you can share.
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).

**Comment by albert-github** [comment]
@hermannkleier  The attachment is missing (attachments through email doesn't work for GitHub).

**Comment by hermannkleier** [comment]
Hope the interface works…
[freopen.tar.gz](https://github.com/user-attachments/files/17554985/freopen.tar.gz)

**Comment by albert-github** [comment]
The interface does work.

> I am wondering if bibtex is needed at all for the HTML output. 

Yes bibtex is needed as the bib entries are translated by means of a perl file and bibtex into a a file that is understood by doxgen (`citelist.doc`) and that also can be used with other output formats through doxygen (like RTF, Docbook).
With the debug option `-d cite` with doxygen intermediate files are left during the generation process and can be inspected.

Depending on the settings of the tex installation the user is prompted or not (in your case) for packages that are not yet installed.

Which packages are missing is hard to tell, but we can see.
The fact that some errors are  given is, in my opinion fortunately, due to the fact that the bibtex cannot be completed properly.

It is hard to tell which packages are missing, and as said before depending on the used texts and also on the used settings (especially the output language).
For the test suite I see for Linux (Ubuntu):
```
        sudo apt update --fix-missing
        sudo apt upgrade
        sudo apt update
        sudo apt-get install texlive texlive-latex-recommended texlive-extra-utils texlive-latex-extra texlive-font-util
````
and for Windows:
```
      uses: teatimeguest/setup-texlive-action@v3
      with:
        packages: >-
          scheme-medium
          collection-latexextra
          babel-dutch
          cjk
          bibtex
```
these  settings might even a bit to large as they are also used to generate the doxygen pdf manual.
